addons/hw_drivers/tools/helpers.py

```diff
@@ -330,3 +330,19 @@ def write_file(filename, text, mode='w'):
         path = path_file(filename)
         with open(path, mode) as f:
             f.write(text)
+
+def download_from_url(download_url, path_to_filename):
+    try:
+        urllib3.request.urlretrieve(download_url, path_to_filename)
+        _logger.info('Downloaded %s from %s', path_to_filename, download_url)
+    except Exception as e:
+        _logger.error('Failed to download from %s: %s', download_url, e)
+
+def unzip_file(path_to_filename, path_to_extract):
```

Comment:
at least this one is ensuring you are working with a zip file but again, not a fan, especially if you remove the file at the end! If used only once, make the unsafe operations in your module.

Reply:
I also replaced the "os.remove" by a call to a different helper (unlink_file)

Reply:
It is not necessary to make the partition writable on Windows because Odoo will have write rights for its own repo. This is the reason ```writable``` [here](https://github.com/odoo/odoo/blob/3a2fd05026c0b0ae7fb852f59d1a74f0fc3db3b2/addons/hw_drivers/tools/helpers.py#L45) is defined as ```nullcontext``` on Windows but not on Raspberry Pi.

I see what you mean by multiple path validity verification, what do you suggest as alternative? Directly call ```Path(path_to_filename).unlink()```?

Reply:
Replaced by a direct call to Path(path_to_filename).unlink()

Reply:
Indeed and done,  I added ```with writable()``` to ```unzip_file```